javascript/features/death_match/death_match_commands.js

```diff
@@ -21,9 +23,19 @@ export class DeathMatchCommands {
     }
 
     // The death match command is being used.
-    onDmCommand(player, zone) { 
-        if(zone === null || zone === undefined || typeof zone !== 'number') {
-            player.sendMessage(Message.DEATH_MATCH_INVALID_ZONE, zone);
+    async onDmCommand(player, zone) { 
+        if(isNaN(zone)) {
```

Comment:
One of the reasons that I suggested that the verification of input `goToDmZone` does right now should move to `death_match_commands.js`, is because you're now creating a tri-state out of it:

  * `/deathmatch`: displays a dialog
  * `/deathmatch 1`: goes to a zone
  * `/deathmatch 123`: shows an error message

While this is fine on itself, it's inconsistent with all other dialog-based commands that we have, which show the dialog for incorrect input as well.

Code's much easier to reason about when all input validation is done in one place (e.g. the player command), after which the parameters are presumed to be correct unless they're coding errors.

Reply:
Agreed, makes things easier inded.
Also agreed on validating user input at the command.